java-checks/src/main/java/org/sonar/java/checks/unused/UnusedMethodParameterCheck.java

```diff
@@ -76,7 +76,11 @@ public void visitNode(Tree tree) {
   }
 
   private static boolean isExcluded(MethodTree tree) {
-    return ((MethodTreeImpl) tree).isMainMethod() || isOverriding(tree) || isSerializableMethod(tree) || isDesignedForExtension(tree);
+    return ((MethodTreeImpl) tree).isMainMethod() || isAnnotated(tree) || isOverriding(tree) || isSerializableMethod(tree) || isDesignedForExtension(tree);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1067)